src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -63,14 +66,30 @@ private static String getRootURL(@NonNull Run<?, ?> build) {
         }
 
         String url = DisplayURLProvider.get().getRunURL(build);
+        return checkURL(url);
+    }
 
-        if (url.startsWith("http://localhost")) {
-            throw new IllegalStateException("Jenkins URL cannot start with http://localhost");
+    /**
+     * Check if the build URL is compatible with Bitbucket API.
+     * For example, Bitbucket API doesn't accept simple hostnames as URLs host value
+     * Throws an IllegalStateException if it is not valid, or return the url otherwise
+     *
+     * @param url the URL of the build to check
+     * @return the url if it is valid
+     */
+    static String checkURL(String url) {
+        /*  */
+        try {
+            URL u = new URL(url);
```

Comment:
Do we really need to cast it to URL? They already did that several times before we use it as a String.

Reply:
Bugger they just rely on strings: https://github.com/jenkinsci/jenkins/blob/44363acab454bee8868206479f0976b837c62ea9/core/src/main/java/jenkins/model/JenkinsLocationConfiguration.java#L162-L166

Reply:
I think this is still the best way to get a correct host from a URL.
I don't think using regex it better and sadly, we can not check only the dot presence as it is possible to have a dot in the rest of the path.

Reply:
Not your fault 😩